consensus/hotstuff/verification/combined_signer_v2.go

```diff
@@ -0,0 +1,123 @@
+package verification
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/consensus/hotstuff/signature"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+)
+
+// CombinedSignerV2 creates votes for the main consensus. Per protocol specification
+// a consensus participant can vote for a block by _either_ signing the block with
+// its staking key _or_ with its random beacon key.
+// If the participant provides signs with its random beacon key, it contributes to
+// hotstuff _and_ the random beacon. A staking signature _only_ contributes to
+// hotstuff. The latter is a useful fallback for a node that failed to properly
+// participate in the DKG.
+// The default behaviour of a node is to contribute to both the random beacon and
+// hotstuff, as it will yield higher rewards. Therefore, we always sign with the
+// random beacon key it is available.
+type CombinedSignerV2 struct {
+	staking              module.AggregatingSigner
+	thresholdSignerStore module.ThresholdSignerStore
+	signerID             flow.Identifier
+}
+
+// NewCombinedSignerV2 creates a new combined signer with the given dependencies:
+// - the hotstuff committee's state is used to retrieve public keys for signers;
+// - the staking signer is used to create and verify aggregatable signatures for the first signature part;
+// - the thresholdVerifier is used to verify threshold signatures
+// - the thresholdSignerStore is used to get threshold-signers by epoch/view;
+// - the signer ID is used as the identity when creating signatures;
+func NewCombinedSignerV2(
+	committee hotstuff.Committee,
+	staking module.AggregatingSigner,
+	thresholdVerifier module.ThresholdVerifier,
```

Comment:
This seems to be used only for testing. I suggest to remove it from the constructor.